bundles/org.eclipse.cdt.lsp.editor.ui.test/src/org/eclipse/cdt/lsp/editor/ui/test/preference/LspEditorPreferencesTesterTest.java

```diff
@@ -0,0 +1,144 @@
+package org.eclipse.cdt.lsp.editor.ui.test.preference;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.eclipse.cdt.lsp.LspPlugin;
+import org.eclipse.cdt.lsp.editor.ui.test.TestUtils;
+import org.eclipse.cdt.lsp.server.ICLanguageServerProvider;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LspEditorPreferencesTesterTest {
+	private static final String LSP_CEDITOR_ID = "org.eclipse.cdt.lsp.CEditor"; //$NON-NLS-1$
+	private static final String C_EDITOR_ID = "org.eclipse.cdt.ui.editor.CEditor"; //$NON-NLS-1$
+	private static final String FILE_CONTENT = "// sample file content";
+	private static final String MAIN_CPP = "main.cpp";
+	private static final String EXTERNAL_HEADER_HPP = "ExternalHeader.hpp";
+	private IProject project;
+	
+	@Before
+	public void setUp() throws CoreException {
+		project = TestUtils.createCProject("LspCEditorTest"+System.currentTimeMillis());
```

Comment:
This project needs to be deleted at the end of the test. CDT's tests have BaseUITestCase5 and BaseTestCase5 which does a bunch of cleanup. I don't actually recommend reusing those base test classes directly as they will tie in years of accumulated "stuff". But the ["assertWorkspaceIsEmpty"](https://github.com/eclipse-cdt/cdt/blob/ab721eed87089d61e4ce4724646a1b1d3a79a484/core/org.eclipse.cdt.core.tests/suite/org/eclipse/cdt/core/testplugin/util/BaseTestCase5.java#L148-L175) I added recently helps ensure that tests are less dependent on each other by failing tests that don't cleanup after themselves.

Reply:
see comment below.